petastorm/etl/dataset_metadata.py

```diff
@@ -77,7 +76,10 @@ def materialize_dataset(spark, dataset_url, schema, row_group_size_mb=None):
         validate_schema=False)
 
     _generate_unischema_metadata(dataset, schema)
-    if not dataset.metadata_path:
+    try:
```

Comment:
Another option here is to check if the _metadata file was generated and automatically using the legacy method if not. I think its best to avoid that for now though unless it becomes a real problem.